docs/tools/mbed_cli_2/migration_guide.md

```diff
@@ -0,0 +1,198 @@
+- [MCUXpresso](https://github.com/ARMmbed/mbed-tools/issues/253)
+- [NetBeans](https://github.com/ARMmbed/mbed-tools/issues/254)
+- [STM32CubeIDE](https://github.com/ARMmbed/mbed-tools/issues/257)
+- [Keil uVision](https://github.com/ARMmbed/mbed-tools/issues/256)
+
+### Library management
+
+<span style="background-color:#ffcccc;">The subcommands `mbed-cli add` (add
+library from URL) and `mbed-cli remove` (remove library) are deprecated.
+Instead, use `git clone` or `mbed-tools import` to clone a library. Then,
+manually create a `reponame.lib` file that contains a single line in the form
+`https://github.com/ARMmbed/reponame#branch-or-tag` to fetch a specific branch
+or tag. If you want to fetch the default branch, you don't need to add
+`#branch-or-tag-name`.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli new --library`
+(force creation of an Mbed library) is deprecated. Create a new Mbed library by
+creating a new folder and adding it to git version control. Then, create a
+`reponame.lib` in the application you wish to use the new library from, as
+described above (in the `mbed-cli add` paragraph).</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli ls` (view
+dependency tree) is deprecated. The subcommand `mbed-tools deploy` will now
+display all dependencies.</span><br>
+
+### Tool configuration
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli config` (tool
+configuration) is deprecated. There is no need to configure an Mbed-specific
+compiler path in Mbed CLI 2; the compiler path is determined in a standard way
+by CMake instead. Other configuration options, like `target`, `toolchain`,
+`protocol`, `depth` and `cache`, are not supported in Mbed CLI 2.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli toolchain`
+(set or get default toolchain) is deprecated. Pass the desired toolchain on the
+command line when compiling with `mbed-tools compile --toolchain GCC_ARM`, to,
+for example, select `GCC_ARM` as the toolchain.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli target` (set
+or get default target) is deprecated. Pass the desired target on the command
+line when compiling with `mbed-tools compile --mbed-target
+DISCO_L475VG_IOT01A`, to, for example, select `DISCO_L475VG_IOT01A` as the
+Mbed target.</span><br>
+
+
+### Creating an Mbed program
+<span style="background-color:#ffe6cc;">The subcommand `mbed-tools new` creates
+a new Mbed program by default and supports only one command-line-option
+`--create-only` (create a program without fetching mbed-os). The following
+command-line options, previously supported in Mbed CLI 1, are deprecated:
+
+* `--scm`: We now always use git.
+* `--program`: There is no need to specify this. We always assume a program.
+* `--library`: There is no need to specify this. We always assume a program.
```

Comment:
Well, this is actually controlled by CMake, rather than what we "assume":
* An application's `CMakeLists.txt` should call `add_executable(<app_name>)` to build an executable and `mbed_set_post_build()` to generate a flashable binary (.bin or .hex or both)
* A library's `CMakeLists.txt` call `add_library(<lib_name)` _without_ calling `mbed_set_post_build()` to build a static library (.a). This is sort of like `--library` of the old tools.

I've experimented that it's totally fine to build only libraries with our tools without an application, as long as the CMake definitions are correct.

Reply:
This is correct and I missed it on my last round of reviews. We don't even "assume a program" in the tool, we assume CMake can generate Ninja build system instructions and the "all" target can be built. We should just say something like "set this in CMake" and perhaps link to a tutorial for configuring a library target.

Reply:
`mbed-tools new` doesn't handle this difference and always assumes you want to make a program. We have one template, and it always makes an application. I'll add instructions on how to modify CMake to create a library.